conformance/lang/values-types-and-variables/map_types.balt

```diff
@@ -0,0 +1,211 @@
+    testMap(a);
+
+    testMap({a: "hello", b: ""});
+}
+
+function testMap(map<string> m) {
+    io:println(m); // @output {"a":"1"}
+                   // @output {"a":"hello","b":""}
+}
+
+Test-case: error
+Description: Test a type map<T> not containing a mapping shape m if a field shape in m has a value shape 
+             that is not in T.
+Labels: map-type-descriptor, mapping-type-descriptor, mapping-constructor-expr, int-type-descriptor, 
+        int-literal, string-literal
+
+public function main() {
+    map<int> a = {a: 1, b: "hello"}; // @error an int map cannot contain string values
+
+    testMap({
+        a: "hello", // @error an int map cannot contain string values 
+        b: "world", // @error an int map cannot contain string values
+        c: 0
+    });
+}
+
+function testMap(map<int> m) {
+}
+
+Test-case: output
+Description: Test the type of the values in the iteration sequence of a value belonging to map<T> being T and the 
+             iteration completion value being nil.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, mapping-constructor-expr, int-literal, 
+        foreach-stmt
+
+public function main() {
+    map<int> a = {a: 1, b: 2};
+
+    foreach int i in a {
+        testNextValue(i);
+    }
+
+    map<int> b = {a: 0};
+    foreach int i in b {
+        testNextValue(i);
+    }
+
+    map<int> c = {};
+    foreach int i in c {
+        testNextValue(i);
+    }
+}
+
+function testNextValue(int i) {
+    io:println(i); // @output 1
+                   // @output 2
+                   // @output 0 
+}
+
+Test-case: error
+Description: Test the type of the values in the iteration sequence of a value belonging to map<T> being T 
+             via incompatible assignments.
+Labels: map-type-descriptor, mapping-type-descriptor, string-type-descriptor, int-type-descriptor, 
+        mapping-constructor-expr, int-literal, foreach-stmt
+
+public function main() {
+    map<int> a = {a: 1, b: 2};
+
+    foreach string s in a { // @error value in the iteration sequence should be int
+
+    }
+}
+
+Test-case: error
+Description: Test the iteration completion value of a map being nil via incompatible assignments.
+Labels: map-type-descriptor, mapping-type-descriptor, record-type-descriptor, int-type-descriptor, 
+        error-type-descriptor, mapping-constructor-expr, lang.map
+
+public function main() {
+    map<int> a = {};    
+    record {| int value; |}|error next = a.iterator().next(); // @error completion type for map iteration is nil
+}
+
+Test-case: output
+Description: Test mapping type-descriptor being covariant in the types of their members.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, byte-type-descriptor, 
+        anydata-type-descriptor, mapping-constructor-expr, int-literal
+
+public function main() {
+    map<byte> a = {a: 1, b: 2};
+    testMap(a);
+
+    map<int> b = a;
+    testMap(b);
+}
+
+function testMap(map<anydata> m) {
```

Comment:
anydata is a complicated type.  If you are not testing `anydata` specifically, I would just use `any`.